src/platform/Zephyr/BLEAdvertisingArbiter.cpp

```diff
@@ -72,6 +80,28 @@ CHIP_ERROR RestartAdvertising()
     return System::MapErrorZephyr(result);
 }
 
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+    .disconnected = [](struct bt_conn * conn, uint8_t reason) { sWasDisconnection = true; },
+    .recycled =
+        []() {
+            // In this callback the connection object was returned to the pool and we can try to re-start connectable
+            // advertising, but only if the disconnection was detected.
```

Comment:
Just curious - in which cases where the connection is recycled we don't want to retry the advertising?

Reply:
This callback is called not only if the disconnection occurs, but on every action that the conn object is released. So, in the case of ChipOverBLE advertising, this callback has been called many times, but actually, there is no need to retry advertising. They deprecated auto-restart on disconnection, so we should take into account only calls if a disconnect occurs to keep previous functionality.